File: examples/docs-code-snippets/src/main/java/examples/java/recipes/RecipeFireEventAndBlock.java
package examples.java.recipes;

import com.ing.baker.recipe.javadsl.InteractionFailureStrategy;
import com.ing.baker.recipe.javadsl.Recipe;

public class RecipeFireEventAndBlock {

    public final static Recipe recipe = new Recipe("example")
        .withDefaultFailureStrategy(
            InteractionFailureStrategy.FireEventAndBlock("MyEvent")
        );
}

File: examples/docs-code-snippets/src/main/scala/examples/scala/recipes/RecipeFireEventAndBlock.scala
package examples.scala.recipes

import com.ing.baker.recipe.common.InteractionFailureStrategy
import com.ing.baker.recipe.scaladsl.Recipe

object RecipeFireEventAndBlock {
  val recipe: Recipe = Recipe("example")
    .withDefaultFailureStrategy(
      InteractionFailureStrategy.FireEventAndBlock(Some("MyEvent"))
    )
}